storm-core/src/jvm/backtype/storm/scheduler/resource/strategies/scheduling/DefaultResourceAwareStrategy.java

```diff
@@ -97,7 +101,7 @@ public ResourceAwareStrategy(Cluster cluster, Topologies topologies) {
 
         if (spouts.size() == 0) {
             LOG.error("Cannot find a Spout!");
-            return null;
+            return SchedulingResult.failure(SchedulingStatus.FAIL_INVALID_TOPOLOGY, "Cannot find a Spout!");
```

Comment:
Well currently there is no early check in nimbus or the client for that.  Technically, a user can submit a topology without a spout.  The DefaultResourceAwareStrategy starts traversing the topology graph starting from the spout thus we need a topology with a spout.  And it doesn't really make since for a topology not to have a spout.  Perhaps we can file a jira a look into putting additional checks for this in the client or on nimbus in the future.